### ISSUE TITLE
read_file: IllegalArgumentException: point array must contain 0 or >1 elements

### ISSUE THREAD

**Issue by IngJuanMaSuarez** [geopandas/geopandas#2791]
### Describe the problem

Hi everyone.

When i try to read this [shapefile](https://drive.google.com/file/d/1GKbleAQ-pQVca_04VeweuJRdXtg5zPiU/view?usp=sharing), i getting this error:

```
File c:\Users\vpt\anaconda3\envs\geo_env\Lib\site-packages\geopandas\io\file.py:259, in _read_file(filename, bbox, mask, rows, engine, **kwargs)
    256     path_or_bytes = filename
    258 if engine == "fiona":
--> 259     return _read_file_fiona(
    260         path_or_bytes, from_bytes, bbox=bbox, mask=mask, rows=rows, **kwargs
    261     )
    262 elif engine == "pyogrio":
...
    121 else:
    122     return simple_geometries_1d(coords, indices, GeometryType.LINESTRING, out=out)

GEOSException: IllegalArgumentException: point array must contain 0 or >1 elements
```

I read this shapefile in a software GIS without problem. That file has reference system and i can to visualize in QGIS. But in GeoPandas i getting this error.

### Steps to reproduce

```
import geopandas as gpd

pluto_exploded = gpd.read_file(r"C:\Users\vpt\Desktop\zlvas\shapefiles\10.lot_line_type\mappluto_exploded_v2\mappluto_exploded_v2_proj.shp")
```

### Versions of your packages

```
geopandas = 0.12.2
geopandas-base = 0.12.2
```

### Your operating system

Windows 11 Home Single Language

**Comment by jdmcbr** [comment]
@IngJuanMaSuarez I took a look at this, and it seems as though QGIS is silently skipping over invalid geometries within the shapefile, whereas geopandas is raising an error. If I add in exception handling (really, exception ignoring) in geopandas, I see the same 4735797 features that appear in QGIS. The 7 features in the file that are skipped are not valid linestrings, and shapefiles don't support mixed geometry types. The simplistic exception catching I added to see this is not something we'd want to add generally, I think, since it could result in mysteriously dropped data. I'm curious, what's the origin of your shapefile that's hitting this issue?

**Comment by koettert** [comment]
@jdmcbr we also recently experienced the problem when reading this [file](https://osdatahub.os.uk/downloads/open/OpenUSRN). Adding an option to the [read_file](https://geopandas.org/en/stable/docs/reference/api/geopandas.read_file.html#geopandas.read_file) method to ignore invalid geometries would be super helpful.

**Comment by martinfleis** [comment]
I agree that it can sometimes be useful to intentionally ignore invalid geometries. @brendan-ward how feasible would that be with pyogrio?

**Comment by brendan-ward** [comment]
On the pyogrio side, it would be just a matter of adding another keyword and passing that that to the `geopandas.array::from_wkb` function.  However, we'd first have to add support on the GeoPandas side to support the underlying shapely [`on_invalid` parameter](https://shapely.readthedocs.io/en/stable/reference/shapely.from_wkb.html#shapely-from-wkb) (which also exists for pygeos).

Added #2962 to track that.

Without further handling, if we expose this in pyogrio, the invalid geometries would be returned as `None` instead of filtered out.  I think that is OK?  That way the user can still get at the other attributes, and would be responsible for screening out those rows.

**Comment by martinfleis** [comment]
I guess that returning None and warning that it has happened and why may be actually a better option than skipping rows. You can easily filter that afterwards and get all you can possible get from a corrupted file.

**Comment by theroggy** [comment]
Adding the `on_invalid` parameter has been merged, so it will be available in the next major version of geopandas. We'll have to wait for that to add support to pyogrio (or if urgent, we might be able to use shapely directly in pyogrio till geopandas 1.0 is released).

However, I was wondering, if we go for returning None for the invalid polygons and warning that there are cases like that in the file, do we still need an extra keyword in `read_file` for it? Are there use cases where getting the error is explicitly better than warning + None values?

**Comment by martinfleis** [comment]
> Are there use cases where getting the error is explicitly better than warning + None values?

Yes, within a long automatised pipeline you may prefer a hard fail rather than risking erroneous result because the warning was not caught.

**Comment by theroggy** [comment]
> > Are there use cases where getting the error is explicitly better than warning + None values?
> 
> Yes, within a long automatised pipeline you may prefer a hard fail rather than risking erroneous result because the warning was not caught.

In general, indeed... I'l just wondering how useful it is in this case, but a keyword more or less probably isn't something to think too much about...

**Comment by EwoutH** [comment]
I have a similar issue: https://github.com/geopandas/geopandas/discussions/3336

**Comment by theroggy** [comment]
I created a PR to add support for this in pyogrio: https://github.com/geopandas/pyogrio/pull/422

It must be noted though that GEOS/Shapely `from_wkb` is a bit more aggressive than e.g. QGIS is for these features. When `on_invalid` "warn" or "ignore" is used with `from_wkb`, the result of any invalid geometry is always None.
QGIS tries to salvage the data available, so e.g. for the geometries I tested with (in https://github.com/geopandas/geopandas/discussions/3336), there is e.g. a geometry that is a multipolygon with one ring only having only 2 points, and another perfectly valid one. So, after doing a make_valid, the one ring is usable.

**Comment by theroggy** [comment]
[pyogrio 0.9](https://github.com/geopandas/pyogrio/releases/tag/v0.9.0) has been released and includes support for a new  [`on_invalid`](https://pyogrio.readthedocs.io/en/latest/api.html#pyogrio.read_dataframe:~:text=on_invalidstr%2C%20optional%20(default%3A%20%E2%80%9Craise%E2%80%9D)) keyword in `read_dataframe`.

As [`geopandas.read_file`](https://geopandas.org/en/stable/docs/reference/api/geopandas.read_file.html) in practice passes all keywords to `pyogrio.read_dataframe` when the `pyogrio` engine is used (the default since geopandas 1.0), you can just paas the `on_invalid` parameter there as well.

So I think this issue can be closed?

**Comment by martinfleis** [comment]
Might be just worth adding a bit to documentation somewhere.

**Comment by EwoutH** [comment]
That’s great news!

Maybe just add a list with possible keyword arguments and an URL to the full documentation?
